torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1433,6 +1433,37 @@ class ShapePropagator {
           dtype, device, dim, /*requires_grad=*/c10::nullopt)};
     };
 
+    static const auto factory_like_with_ndim = [](Node* node,
+                                                  int dim) -> type_vec_t {
+      at::optional<IValue> maybe_layout_option = node->get(attr::layout);
+      if (!maybe_layout_option)
+        return {};
+
+      at::optional<IValue> maybe_device_option = node->get(attr::device);
+      if (!maybe_device_option)
+        return {};
+
+      auto tt = node->input(0)->type()->expect<TensorType>();
+
+      auto in_type = tt->scalarType();
```

Comment:
similarly, could you move the declaration of this down to the use ? and call it `in_dtype`?

Reply:
👍